circuits/lib/query/credentialAtomicQuerySigOnChainSmt.circom

```diff
@@ -0,0 +1,238 @@
+    signal input issuerAuthRevTreeRoot;
+    signal input issuerAuthRootsTreeRoot;
+    signal output issuerAuthState;
+
+    // issuerClaim non rev inputs
+    signal input issuerClaimNonRevMtp[IssuerLevels];
+    signal input issuerClaimNonRevMtpNoAux;
+    signal input issuerClaimNonRevMtpAuxHi;
+    signal input issuerClaimNonRevMtpAuxHv;
+    signal input issuerClaimNonRevClaimsTreeRoot;
+    signal input issuerClaimNonRevRevTreeRoot;
+    signal input issuerClaimNonRevRootsTreeRoot;
+    signal input issuerClaimNonRevState;
+
+    /* current time */
+    signal input timestamp;
+
+    /** Query */
+    signal input claimSchema;
+    signal input slotIndex;
+    signal input operator;
+    signal input value[valueArraySize];
+
+
+    /*
+    >>>>>>>>>>>>>>>>>>>>>>>>>>> End Inputs <<<<<<<<<<<<<<<<<<<<<<<<<<<<
+    */
+
+    /* Id ownership check*/
+    component userIdOwnership = IdOwnershipBySignatureOnChainSmt(IdOwnershipLevels, OnChainSmtLevels);
+
+    userIdOwnership.userID <== userID;
+    userIdOwnership.userState <== userState;
+
+    userIdOwnership.userClaimsTreeRoot <== userClaimsTreeRoot; // currentHolderStateClaimsTreeRoot
+    for (var i=0; i<IdOwnershipLevels; i++) { userIdOwnership.userAuthClaimMtp[i] <== userAuthClaimMtp[i]; }
+    for (var i=0; i<8; i++) { userIdOwnership.userAuthClaim[i] <== userAuthClaim[i]; }
+
+    userIdOwnership.userRevTreeRoot <== userRevTreeRoot;  // currentHolderStateClaimsRevTreeRoot
+    for (var i=0; i<IdOwnershipLevels; i++) { userIdOwnership.userAuthClaimNonRevMtp[i] <== userAuthClaimNonRevMtp[i]; }
+    userIdOwnership.userAuthClaimNonRevMtpNoAux <== userAuthClaimNonRevMtpNoAux;
+    userIdOwnership.userAuthClaimNonRevMtpAuxHv <== userAuthClaimNonRevMtpAuxHv;
+    userIdOwnership.userAuthClaimNonRevMtpAuxHi <== userAuthClaimNonRevMtpAuxHi;
+
+    userIdOwnership.userRootsTreeRoot <== userRootsTreeRoot; // currentHolderStateClaimsRootsTreeRoot
+
+    userIdOwnership.challenge <== challenge;
+    userIdOwnership.challengeSignatureR8x <== challengeSignatureR8x;
+    userIdOwnership.challengeSignatureR8y <== challengeSignatureR8y;
+    userIdOwnership.challengeSignatureS <== challengeSignatureS;
+
+    userIdOwnership.userStateInOnChainSmtRoot <== userStateInOnChainSmtRoot;
+    for (var i=0; i<OnChainSmtLevels; i++) { userIdOwnership.userStateInOnChainSmtMtp[i] <== userStateInOnChainSmtMtp[i]; }
+    userIdOwnership.userStateInOnChainSmtMtpAuxHi <== userStateInOnChainSmtMtpAuxHi;
+    userIdOwnership.userStateInOnChainSmtMtpAuxHv <== userStateInOnChainSmtMtpAuxHv;
+    userIdOwnership.userStateInOnChainSmtMtpNoAux <== userStateInOnChainSmtMtpNoAux;
+
+    userIdOwnership.userCorrelationID <== userCorrelationID;
+    userIdOwnership.userNullifier <== userNullifier;
+
+
+    // Check issuerClaim is issued to provided identity
+    component claimIdCheck = verifyCredentialSubject();
+    for (var i=0; i<8; i++) { claimIdCheck.claim[i] <== issuerClaim[i]; }
+    claimIdCheck.id <== userID;
```

Comment:
if claim will be issued on nullifer  how it will affect this check?

Reply:
Good point. It will change (or added as OR option via mux) to take into account the nullifier calculation. Kind of `hash(userID, salt) === nullifier` Something may be needed to support it from user DB organisation side, which will be considered in a separate task.